### DISCUSSION TITLE
Big list of observations I made: Missed Features, Bugs & Spicy Suggestions

### DISCUSSION THREAD

**Discussion by yuuiko** [ppy/osu#22073] [Ideas] [answered]
TOP TOOLBAR
* Toolbar → Username → Click Flag. This opens the rankings panel?
* Clock: Replace Second Hand with Pie Graph - Duration Client Open. Could mimic the hour hand: Start of pie would be the hour began, and fill forward to the current hour.
* PP / rank indication like stable on song select. Probably under the username
* Match osudirect icon in toolbar with browse button in main menu or other way around (tbh i like the main menu icon better)
* Shuffle button for Music Player. Right now it just copies the current song select order #13039 
* Click profile name: The card that shows should match discord rich presence for consistency

SONG SELECT
* **Mania: Keycount indication on Beatmap difficulty items (e.g. 4K, 5K, 7K)**
* **Return to main menu after idle on song select so music can progress instead of looping and making you tired of the selected song.**
* **No way to manage and add to collections with touch**
* View a collection, Right click a map, press delete: Add a “remove from collection” button alongside delete beatmap. I was deleting beatmaps thinking i was just removing them from collection.
* Remember sorting option and sort by menu configuration across game close
* Group functions from stable are missing
* Click beatmap name wedge to view fullscreen artwork and download (to use as wallpaper etc.) #13031

MODS / RANDOM / OPTIONS BAR
* **Hold shift and click mods button to “revert mods”: Mods selected: Clear mods. Shift+mods again to recover mods. Inspired by the rewind random function**
* Mods: Remember mods after game restart and through switching modes
* Options: "Removed from unplayed" button doesn’t really make sense. Revise Label: “mark as played / unplayed” 

* **New Mod: Training Mode.** Basically no fail with the ability to scroll the beatmap to practice a certain time. Or just add this to no fail, or make it automatically rewind when you fail / changeable parameters (reset when achieved fail, meh, good) to try again

LEGACY SKINS
Check this skin: https://osu.ppy.sh/community/forums/topics/504196?n=1
* CATCH: Platter scaling Issue with Nightgate Platter
* TAIKO: Taiko drum rim was jank (reversed) on stable and isn’t jank on lazer

Check this skin: https://osu.ppy.sh/community/forums/topics/471776?n=1
* STD: Large animations on Voez skin causes significant lag (android, galaxy fold 4)
* STD: Voez skin hitcircles are not animating properly. Supposed to be according to hitcircle shown duration, slider start is not animating at all

MULTI
* Any Game: Clicking a beatmap should play the web preview clip if the song isn’t downloaded yet
* **Any Game: No way to change freemods**
* Playlist Game: indicate which songs you have played
* Lobby: its a little unclear that the mode select in the toolbar influences what is listed. Perhaps just list all games regardless of game mode and add game mode indicators.

PAUSE & FAIL
* **No way to change visual settings after beatmap has started. Stable has the options on pause menu**
* Continue button for failed screen - keep playing unranked with nofail

PLAYFIELD - GENERAL
* **Leaderboard covers taiko playfield**
* Leaderboard can cover standard playfield on non-wide aspect ratios. Render notes above leaderboard?
* Add new osu mascots to autoplay profiles

PLAYFIELD - MANIA
* Add Coloured bar-lines to warn about SV (aka. LN / scroll speed / BPM) changes. (inspiration: Cytus II gameplay)
* Pattern based higlighting for mania. e.g. highlight same-time notes (double, triple, quads) and flicks (really closely timed notes that are almost doubles idk what they are called)
* Combo Colours for Mania?
* Mania Argon: notes render order - later notes on top of previous. Perhaps add an option for this on skin.ini? https://github.com/ppy/osu/discussions/21994

RESULTS
* **PP celebration animation for results panel e.g. 700 / 727pp with a pie or bar graph** 
* **Rank position number for results panel e.g. # 45 of 21485**

UTILITIES
* **Cloud Save maps, collections and local scores - osu!supporter?**

SYSTEM
* MacOS, Windowed: Issues seem to be happening when interacting with window borders while using raw mouse input. The confine mouse to window borders option becomes unstable.

Ask me for any more details about any specific thing.

**Comment by peppy** [comment, marked as answer]
Thanks for your  feedback. Basically everything you've asked for is addressed by other existing issues. so we will track them there.
